Code/Framework/AzToolsFramework/Tests/InstanceDataHierarchy.cpp

```diff
@@ -1316,6 +1316,105 @@ namespace UnitTest
         }
     };
 
+    class InstanceDataHierarchyEndGroupTest
+        : public AllocatorsFixture
+    {
+    public:
+        class EndGroupContainer
+            : public AZ::Component
+        {
+        public:
+            AZ_COMPONENT(EndGroupContainer, "{0CFC7838-9B01-4E25-8932-1C1EE4FCEC77}", AZ::Component);
+
+            int m_rootData;
+            bool m_otherRootData;
+            int m_group1Data;
+            int m_group2Data;
+            int m_finalRootData;
+
+            static void Reflect(AZ::ReflectContext* context)
+            {
+                if (auto serialize = azrtti_cast<AZ::SerializeContext*>(context))
+                {
+                    serialize->Class<EndGroupContainer, AZ::Component>()
+                        ->Field("rootData", &EndGroupContainer::m_rootData)
+                        ->Field("otherRootData", &EndGroupContainer::m_otherRootData)
+                        ->Field("group1Data", &EndGroupContainer::m_group1Data)
+                        ->Field("group2Data", &EndGroupContainer::m_group2Data)
+                        ->Field("finalRootData", &EndGroupContainer::m_finalRootData)
+                        ;
+
+                    if (auto editContext = serialize->GetEditContext())
+                    {
+                        editContext->Class<EndGroupContainer>("EndGroupTest", "")
+                            ->DataElement(nullptr, &EndGroupContainer::m_rootData)
+                            ->DataElement(nullptr, &EndGroupContainer::m_otherRootData)
+
+                            ->ClassElement(AZ::Edit::ClassElements::Group, "First Group")
+                            ->DataElement(nullptr, &EndGroupContainer::m_group1Data)
+
+                            ->ClassElement(AZ::Edit::ClassElements::Group, "Second Group")
+                            ->DataElement(nullptr, &EndGroupContainer::m_group2Data)
+                            ->EndGroup()
+
+                            ->DataElement(nullptr, &EndGroupContainer::m_finalRootData)
+                            ;
+                    }
+                }
+            }
+
+            // AZ::Component overrides ...
+            void Activate() override {}
+            void Deactivate() override {}
+        };
+
+        void run()
+        {
+            using namespace AzToolsFramework;
```

Comment:
nit: Ideally best to avoid `using namespace` in general (even at function scope - there's a goodreads about this 😉) and instead us a namespace alias or a using declaration (e.g. `using AzToolsFramework::InstanceDataHierarchy` etc...)

Reply:
Thanks! Sorry this was from a nearby test copy-pasta, there was only one line using it so I just went ahead and removed it :)